report_gen/README.md

```diff
@@ -0,0 +1,157 @@
+# Report generation
+
+## Using graphgen.py
+
+### Arguments:  
+	-f, --csv_file - CSV file(s) to process ( Required )  
+	-o, --out_dir - Output directory for the graphs ( Default: ./ )  
+	-x, --x_axes_str - Create graphs with x axes being the columns containing these strings ( Default: First column of first file )  
+	--plot_col - Only plot the columns with these names while plotting across csv files ( Default: All columns )  
+	-e, --extension - Specify the extension of the graphs ( Default: svg )  
+	-c, --categories - Plot by category ( Default: no category )   
+
+### Basic functionality
+
+Read supplied csv files, create graphs from the columns they have in common, and store them. As default, the x axis is the first column in the first file.  
+
+Ignores the first line of the csv file, as it commonly holds the name of the command which generated it.
+
+Output filenames are of the format: all:x_axis-y_axis.extension
+
+####CSV files used in example:
+
+    a.csv:                 b.csv:                 c.csv:
+    
+    firstline              firstline              firstline
+    x0, y1, y2, y3         y1, x0, y2, z4         x0, y1, y3, z4
+    1,  2,  8,  4          1,  1,  9,  5          1,  4,  8,  6
+    2,  4,  4,  3          2,  3,  5,  4          2,  6,  4,  5
+    3,  8,  2,  2          4,  7,  3,  3          3,  7,  2,  4
+
+As seen from the structure of the csv files:  
+1. The order of columns can be different from file to file  
+2. Not all files have to contain the same columns
+
+#### Example usage:
```

Comment:
I would prefer the following structure:
## Using graph generator (graphgen.py)
### Arguments
### Basic functionality
### Input files specification
**Input files example**
### Examples of usage
#### Basic usage
**Sample command**
**Sample output**
#### Filter columns
...
#### Custom x axes
...
#### Grouping data from multiple files
#### Output files extension